### PR TITLE
[FLINK-19055] Wait less time for all memory GC in tests (MemoryManager#verifyEmpty)

### PR DESCRIPTION
**Pull request by azagrebin**
#12980 increased the waiting time to GC all allocated memory to verify that nothing is allocated in `MemoryManager`/`UnsafeMemoryBudget`. The waiting happens in a separate thread in production and does not disrupt anything. The problem is that tests use the same method which blocks the testing thread for too long. We do not need such big waiting time in tests. The PR refactors code to get back to the smaller waiting time in tests.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/memory/UnsafeMemoryBudget.java [hunk 1]
```diff
@@ -36,16 +36,20 @@
  */
 class UnsafeMemoryBudget {
 	private static final int MAX_SLEEPS = 11; // 2^11 - 1 = (2 x 1024) - 1 ms ~ 2 s total sleep duration
-	private static final int MAX_SLEEPS_VERIFY_EMPTY = 17; // 2^17 - 1 = (128 x 1024) - 1 ms ~ 2 min total sleep duration
+	static final int MAX_SLEEPS_VERIFY_EMPTY_FOR_TESTS = 10; // 2^10 - 1 = (1 x 1024) - 1 ms ~ 1 s total sleep duration
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/memory/UnsafeMemoryBudget.java:39]: Testing code should not be contained in production code.

File: flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java [hunk 2]
```diff
@@ -89,12 +89,14 @@
 	 *
 	 * @param memorySize The total size of the off-heap memory to be managed by this memory manager.
 	 * @param pageSize The size of the pages handed out by the memory manager.
+	 * @param verifyEmptyWaitGcMaxSleeps defines how long to wait for GC of all allocated memory to check for memory leaks,
+	 *                                   see also {@link UnsafeMemoryBudget}.
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java:93]: Hmm, the description says that it is the time to wait for GC but isn't is the number of exponential retries?

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @azagrebin. The changes look good to me. I had a comment concerning the JavaDoc description which might be misleading and a comment about testing code being contained in production code. Once these comments are addressed feel free to merge the PR.

File: flink-runtime/src/main/java/org/apache/flink/runtime/memory/UnsafeMemoryBudget.java (hunk 1 above)
Comment (azagrebin) [reply flink-runtime/src/main/java/org/apache/flink/runtime/memory/UnsafeMemoryBudget.java:39]: True, I can move to the testing builder

File: flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java (hunk 2 above)
Comment (azagrebin) [reply flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java:93]: You are right, I just did not want to repeat all details here, so I referred to `UnsafeMemoryBudget` for full description.

**Pull request closed without merging**